src/common/transformations/include/ov_ops/rotary_positional_embeddings.hpp

```diff
@@ -27,6 +27,7 @@ class TRANSFORMATIONS_API RoPE : public Op {
         bool is_interleaved = false;   // interleaved mode, implies trans0213 happens after RoPE
         size_t rotary_ndims = 0;       // dimensions to be embedded (d in the description)
         bool is_chatglm = false;       // chatglm is special which overrides other setting
+        bool support_2d_rope = false;      // chatglm4 is special which overrides other setting
```

Comment:
`support_2d_rope` is a little bit ambiguous, the difference between chatglm2/3 and chatglm4 is the order of input/output.

Reply:
there are two different points between chatglm and chatglm4. first is the order of input / output is changed. second, cos_sin_cache does 2 dimensional calculation to support independent positional embedding of batch and head. The reason of transposed input / output data is to support 2 dimentional positional embeding. So I use the name of config.

Reply:
Maybe this comment should be added to explain this config in details.

Reply:
Okay, I see. I will update it. thanks.

Reply:
Update comments, please review it again. thanks.